consensus/polybft/state_store_exit_event.go

```diff
@@ -54,11 +62,17 @@ type ExitEventStore struct {
 }
 
 // initialize creates necessary buckets in DB if they don't already exist
-func (s *ExitEventStore) initialize(tx *bolt.Tx) error {
+func (s *ExitEventStore) initialize(tx *bolt.Tx, blockNumber uint64) error {
 	if _, err := tx.CreateBucketIfNotExists(exitEventsBucket); err != nil {
 		return fmt.Errorf("failed to create bucket=%s: %w", string(exitEventsBucket), err)
 	}
 
+	if forkmanager.GetInstance().IsForkEnabled(chain.DoubleSignSlashing, blockNumber) {
```

Comment:
Why do we need to initialize store through the fork manager? Isn't it enough to initialize it anyway and do not use it if we do not need it?

Reply:
Good point and you are right, storage logic should be as simple as possible.
Fixed
https://github.com/0xPolygon/polygon-edge/pull/1823/commits/429357f227843802a249e8fdb494b950bdfb638c